lib/nostrum/struct/permission.ex

```diff
@@ -0,0 +1,115 @@
+defmodule Nostrum.Struct.Permission do
+  use Bitwise
+
+  @type general_permission ::
+          :create_instant_invite
+          | :kick_members
+          | :ban_members
+          | :administrator
+          | :manage_channels
+          | :manage_guild
+          | :view_audit_log
+          | :view_channel
+          | :change_nickname
+          | :manage_nicknames
+          | :manage_roles
+          | :manage_webhooks
+          | :manage_emojis
+
+  @type text_permission ::
+          :add_reactions
+          | :send_messages
+          | :send_tts_messages
+          | :manage_messages
+          | :embed_links
+          | :attach_files
+          | :read_message_history
+          | :mention_everyone
+          | :use_external_emojis
+
+  @type voice_permission ::
+          :connect
+          | :speak
+          | :mute_members
+          | :deafen_members
+          | :move_members
+          | :use_vad
+
+  @type t ::
+          general_permission
+          | text_permission
+          | voice_permission
+
+  @type permission_set :: MapSet.t(t)
+
+  @doc """
+  Returns a permission set containing all permissions.
+  """
+  @spec all() :: permission_set
+  def all do
+    permission_to_bitvalue_map()
+    |> Enum.map(fn {perm, _} -> perm end)
+    |> MapSet.new()
+  end
+
+  @doc """
+  Converts the given bitset to a permission set.
+
+  ## Examples
+
+  ```Elixir
+  iex> Nostrum.Struct.Permission.from_bitset(0x04000000)
+  #MapSet<[:change_nickname]>
+
+  iex> Nostrum.Struct.Permission.from_bitset(0x08000002)
+  #MapSet<[:kick_members, :manage_nicknames]>
+
+  iex> Nostrum.Struct.Permission.from_bitset(0)
+  #MapSet<[]>
+  ```
+  """
+  @spec from_bitset(integer) :: permission_set
+  def from_bitset(bitset) do
+    Enum.reduce(permission_to_bitvalue_map(), [], fn {perm, bitvalue}, acc ->
+      if band(bitset, bitvalue) == bitvalue do
+        acc ++ [perm]
+      else
+        acc
+      end
+    end)
+    |> MapSet.new()
+  end
+
+  defp permission_to_bitvalue_map do
```

Comment:
Do you think this would be better suited as a module attribute?

Reply:
It looks like using a module attribute results in a 2 second improvement against 1,000,000 `from_bitset/1` calls in the worst case scenario.

Addressed in 4466136.